### ISSUE TITLE
[Hyundai] maxRange is undefined

### ISSUE THREAD

**Issue by milhaus** [athal7/homebridge-hyundai-bluelink#7]
**Describe The Bug:**
This error pops up in the logs every time it checks. I've tried plugging in different values, swapping the order of maxRange and VIN so I could make sure there weren't linefeed/whitespace issues with the value, making it a string so I could append "mi" to it. Nothing works to remove the error. Other integer values in the config are fine. The only thing I get in homekit from the plugin is locks and on/off. Vehicle is a 2021 Kona EV.

**To Reproduce:**
This happens automatically with each check.

**Expected behavior:**
No error in logs. Charge level reported in HomeKit

**Logs:**

```
[2/23/2021, 2:01:53 PM] [Hyundai] maxRange is undefined
[2/23/2021, 2:11:54 PM] [Hyundai] maxRange is undefined
[2/23/2021, 2:21:53 PM] [Hyundai] maxRange is undefined
[2/23/2021, 2:31:53 PM] [Hyundai] maxRange is undefined
```

**Plugin Config:**

```json
{
            "credentials": {
                "username": "[redacted]",
                "password": "[redacted]",
                "region": "US",
                "pin": "[redacted]"
            },
            "vehicles": [
                {
                    "vin": "[redacted]",
                    "maxRange": 258
                }
            ],
            "remoteStart": {
                "cool": false,
                "heat": false,
                "defrost": false,
                "igniOnDuration": 15
            },
            "refreshInterval": 600,
            "platform": "Hyundai"
        }
```

**Environment:**

* **Plugin Version**: 1.2.11
* **Homebridge Version**: 1.3.1 and a few prior
* **Node.js Version**: 14.15.5
* **NPM Version**: 6.14.11
* **Operating System**: Debian Buster

**Comment by athal7** [comment]
Can you run homebridge in debug mode? Want to see what the status update payload looks like:

https://github.com/athal7/homebridge-hyundai-bluelink/blob/7d60d51f1fe770fa5b9cde32d023b7be6dd3f561/src/platformAccessory.ts#L31

**Comment by milhaus** [comment]
Oh, good hunch. Looks like range is undefined in the update payload.

```
[2/23/2021, 6:00:40 PM] [Hyundai] Received status update {
  chassis: {
    hoodOpen: false,
    trunkOpen: false,
    locked: true,
    openDoors: {
      frontRight: false,
      frontLeft: false,
      backLeft: false,
      backRight: false
    },
    tirePressureWarningLamp: {
      rearLeft: false,
      frontLeft: false,
      frontRight: false,
      rearRight: false,
      all: false
    }
  },
  climate: {
    active: false,
    steeringwheelHeat: false,
    sideMirrorHeat: false,
    rearWindowHeat: false,
    defrost: false,
    temperatureSetpoint: 'LO',
    temperatureUnit: 1
  },
  engine: {
    ignition: false,
    adaptiveCruiseControl: false,
    range: undefined,
    charging: false,
    batteryCharge12v: 85,
    batteryChargeHV: 85
  }
}
[2/23/2021, 6:00:40 PM] [Hyundai] Vehicle is Locked
[2/23/2021, 6:00:40 PM] [Hyundai] maxRange is undefined
[2/23/2021, 6:00:40 PM] [Hyundai] Vehicle is Off
```

**Comment by athal7** [comment]
Thanks! We may need to bring this over to the bluelinky library, since the type definition they have claims `engine.range` will always be a number https://github.com/Hacksore/bluelinky/blob/039a0c25c19666fed9c38d06b80c80e56e27e5cd/src/interfaces/common.interfaces.ts#L41

**Comment by milhaus** [comment]
It would be timely. Looks like they're actively working on engine.range right now:

https://github.com/Hacksore/bluelinky/commit/c326c3fe07b156e4f2aec8afc4425185d2fe2002#diff-4bc921cf635defe83396ed691d13ea4c47bd9ed50241618e2cc3a802aa05d03aL274

My first instinct for now on this side would be validation, maybe around:
https://github.com/athal7/homebridge-hyundai-bluelink/blob/7d60d51f1fe770fa5b9cde32d023b7be6dd3f561/src/services/range.ts#L30

But that would probably be premature before cluing bluelinky in because who knows what engine.range value would be in the undefined case once they decide how they'd like to handle it.

**Comment by milhaus** [comment]
#9 does not fix this for me. I confirmed bluelinky in the plugin was v6.0.1 and even went as far as to reboot the server homebridge is running on.

**Comment by milhaus** [comment]
Ran homebridge in debug mode and got the same undefined value in the response payload for range as listed above.

**Comment by athal7** [comment]
@milhaus all set here?

**Comment by milhaus** [comment]
Oddly I'm still getting that error. homebridge-hyundai-bluelink 1v.3.2, node v14.16.1, npm v7.10.0 on debian 10.

I notice bluelinky closed out 106 but it wasn't listed in release notes for 7.x

I just restarted homebridge to get fresh results and got the undefined error. As a sanity check I did an npm list in the plugins node-modules directory and noticed some busted dependencies. So I did an npm update and restarted homebridge and now I'm getting exciting new errors. I don't know if I just messed up all the perms (forgot I was root) and that's the cause of new errors.
